File: findMissingLetter.py
""" 

#Find the missing letter

Write a method that takes an array of consecutive (increasing) letters as input and that returns the missing letter in the array.

You will always get an valid array. And it will be always exactly one letter be missing. The length of the array will always be at least 2.
The array will always contain letters in only one case.

Example:

['a','b','c','d','f'] -> 'e' ['O','Q','R','S'] -> 'P'

["a","b","c","d","f"] -> "e"
["O","Q","R","S"] -> "P"
(Use the English alphabet with 26 letters!)

"""

def find_missing_letter(chars):
    # construct answer keys
    letters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
    letters = [x for x in letters]

    # find chars starting index
    index = letters.index(chars[0])
    
    # loop through answer keys from index for the length of chars
    for i in range(len(chars)):
        if chars[i] != letters[index+i]:
            return letters[index + i]


# Best Solution

def find_missing_letter(chars):
    n = 0
    while ord(chars[n]) == ord(chars[n+1]) - 1:
        n += 1
    return chr(1+ord(chars[n]))

File: findMissingLetter.js
/* #Find the missing letter

Write a method that takes an array of consecutive (increasing) letters as input and that returns the missing letter in the array.

You will always get an valid array. And it will be always exactly one letter be missing. The length of the array will always be at least 2.
The array will always contain letters in only one case.

Example:

['a','b','c','d','f'] -> 'e' ['O','Q','R','S'] -> 'P'

["a","b","c","d","f"] -> "e"
["O","Q","R","S"] -> "P"
(Use the English alphabet with 26 letters!) */


function findMissingLetter(array)
{
  let letters = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('')
  let ind = letters.indexOf(array[0])
  
  for (i=0;i<array.length;i++){
    if (array[i] !== letters[ind+i]) {
      return letters[ind+i]
    }
  }
}

// Best Solution

function findMissingLetter(array)
{
  let i = array[0].charCodeAt(0);
  return String.fromCharCode(array.find(el => el.charCodeAt(0) != i++).charCodeAt(0) - 1);
}
